src/AutoRest.CSharp/LowLevel/Generation/LowLevelConvenienceMethodWriter.cs

```diff
@@ -1,119 +0,0 @@
-            }
-            writer.Line();
-        }
-
-        private void WriteClientConvenienceMethod(string diagnosticsPropertyName, bool async)
-        {
-            using (WriteConvenienceMethodDeclaration(async))
-            {
-                if (_convenienceMethod.Diagnostic != null)
-                {
-                    using (WriteDiagnosticScope(_writer, _convenienceMethod.Diagnostic, diagnosticsPropertyName))
-                    {
-                        WriteClientConvenienceMethodBody(async);
-                    }
-                }
-                else
-                {
-                    WriteClientConvenienceMethodBody(async);
-                }
-            }
-
-            _writer.Line();
-        }
-
-        private CodeWriter.CodeWriterScope WriteConvenienceMethodDeclaration(bool async)
-        {
-            var methodSignature = _convenienceMethod.Signature.WithAsync(async);
-            _writer
-                .WriteMethodDocumentation(methodSignature)
-                .WriteXmlDocumentation("remarks", $"{methodSignature.DescriptionText}");
-            var scope = _writer.WriteMethodDeclaration(methodSignature);
-            if (_convenienceMethod.BodyParameter != null)
-            {
-                _writer.WriteParametersValidation(new[] { _convenienceMethod.BodyParameter });
-            }
-            return scope;
-        }
-
-        private void WriteClientConvenienceMethodBody(bool async)
-        {
-            string contextVariableName = _convenienceMethod.Signature.Parameters.FirstOrDefault(parameter => parameter.Name == KnownParameters.RequestContext.Name) != null ? $"{KnownParameters.RequestContext.Name}1" : KnownParameters.RequestContext.Name;
-            _writer.Line($"{typeof(RequestContext)} {contextVariableName} = FromCancellationToken({KnownParameters.CancellationTokenParameter.Name});");
-
-            string responseVariableName = _convenienceMethod.Signature.Parameters.FirstOrDefault(parameter => parameter.Name == "response") != null ? "response1" : "response";
-            var protocolSignature = _convenienceMethod.LowLevelClientMethod.Signature;
-            _writer.Line($"{typeof(Response)} {responseVariableName} = {(async ? "await " : String.Empty)}{protocolSignature.Name}{(async ? "Async" : String.Empty)}({string.Join(", ", protocolSignature.Parameters.Select(parameter => ParameterToItsName(parameter)))}){(async ? ".ConfigureAwait(false)" : String.Empty)};");
-
-            if (_convenienceMethod.ResponseType == null)
-            {
-                _writer.Line($"return {responseVariableName};");
-            }
-            else
-            {
-                _writer.Line($"return Response.FromValue({_convenienceMethod.ResponseType}.FromResponse({responseVariableName}), {responseVariableName});");
-            }
-
-            string ParameterToItsName(Parameter parameter)
-            {
-                if (parameter.Name == KnownParameters.RequestContext.Name)
```

Comment:
@pshao25 Ideally, we should compare references to parameters, i.e.: `parameter == KnownParameters.RequestContext`. If it isn't possible (e.g. in case of endpoint), we need to compare `Name` and `Type` (with or without nullability, depending on context).